### PR TITLE
Create BaseX.php

### PR DESCRIPTION
**Pull request by Sevada797**
This function let's you calculate base X of any number <11

### CODE REVIEW

**Review by darwinz — CHANGES_REQUESTED** [review]
Thanks for wanting to contribute.  Please read the contributing guide for making contributions to this repo found at [CONTRIBUTING.md](https://github.com/TheAlgorithms/PHP/blob/master/CONTRIBUTING.md)

**Comment by Sevada797** [comment]
Hi @darwinz thanks for checking my commit, I have read the file [CONTRIBUTING.md](https://github.com/TheAlgorithms/PHP/blob/master/CONTRIBUTING.md), I think everything is OK, if possible please tell me what exact changes should I make.

**Comment by darwinz** [comment]
Please follow modern PHP style guide rules.  We generally stick to the guidelines in the [PSR-12 style guide](https://xoops.gitbook.io/xoops-modules-cookbook/coding-standards/standards/styleguide).  Code should be in a class with a function that defines the algorithm.  You also need to add unit tests.  Format your code so that it's readable and maintainable using the PSR-12 style guide.  Thanks!

**Comment by Sevada797** [comment]
> Please follow modern PHP style guide rules. We generally stick to the guidelines in the [PSR-12 style guide](https://xoops.gitbook.io/xoops-modules-cookbook/coding-standards/standards/styleguide). Code should be in a class or function that defines the algorithm. You also need to add unit tests. Format your code so that it's readable and maintainable using the PSR-12 style guide. Thanks!

I've done it, can you please check it, sorry for the long delay.

File: DIRECTORY.md [hunk 1]
```diff
@@ -29,6 +29,7 @@
   * [Absolutemax](./Maths/AbsoluteMax.php)
   * [Absolutemin](./Maths/AbsoluteMin.php)
   * [Armstrongnumber](./Maths/ArmstrongNumber.php)
+  * [BaseX](./BaseX.php)
```
Comment (darwinz) [inline DIRECTORY.md:32]:
```suggestion
  * [BaseX](./Maths/BaseX.php)
```

**Review by darwinz — COMMENTED** [review]
This looks so much better.  Thanks for updating!  Please see one small change I requested, and also please add a unit test for this algorithm.  Thanks!

**Comment by Sevada797** [comment]
Hi @darwinz, can you see if everything is okay now please?

**Review by darwinz — APPROVED** [review]

File: tests/Maths/MathsTest.php [hunk 2]
```diff
@@ -19,9 +19,18 @@
...
 
 class MathsTest extends TestCase
 {
+
+    public function testBaseX() {
+    $this->assertEquals(11, baseX(3, 2));
+    $this->assertEquals(22, baseX(8, 3));   
+    $this->assertEquals(21, baseX(15, 7));
+    $this->assertEquals(20, baseX(16, 8));
+}
```
Comment (darwinz) [inline tests/Maths/MathsTest.php:33]:
```suggestion
    public function testBaseX() 
    {
        $this->assertEquals(11, baseX(3, 2));
        $this->assertEquals(22, baseX(8, 3));   
        $this->assertEquals(21, baseX(15, 7));
        $this->assertEquals(20, baseX(16, 8));
    }
```

**Review by darwinz — CHANGES_REQUESTED** [review]
Approved on accident.  There are still some formatting issues.  One way to check before committing is to run

```
vendor/bin/phpcs -n
```

File: tests/Maths/MathsTest.php [hunk 3]
```diff
@@ -19,9 +19,18 @@
 require_once __DIR__ . '/../../Maths/Median.php';
 require_once __DIR__ . '/../../Maths/Mode.php';
 require_once __DIR__ . '/../../Maths/FastInverseSquareRoot.php';
+require_once __DIR__ . '/../../Maths/BaseX.php';
 
 class MathsTest extends TestCase
 {
+    public function testBaseX()
```
Comment (darwinz) [inline tests/Maths/MathsTest.php:26]:
```suggestion
    public function testBaseX()
```

File: tests/Maths/MathsTest.php [hunk 4]
```diff
@@ -19,9 +19,18 @@
...
 require_once __DIR__ . '/../../Maths/Mode.php';
 require_once __DIR__ . '/../../Maths/FastInverseSquareRoot.php';
+require_once __DIR__ . '/../../Maths/BaseX.php';
 
 class MathsTest extends TestCase
 {
+    public function testBaseX() 
+    {
+        $this->assertEquals(11, baseX(3, 2));
+        $this->assertEquals(22, baseX(8, 3));
```
Comment (darwinz) [inline tests/Maths/MathsTest.php:29]:
```suggestion
        $this->assertEquals(22, baseX(8, 3));
```

**Review by darwinz — APPROVED** [review]
Thank you!

**Pull request merged**